addons/email_template/email_template.py

```diff
@@ -505,6 +505,11 @@ def generate_email_batch(self, cr, uid, template_id, res_ids, context=None, fiel
             # Add report in attachments: generate once for all template_res_ids
             if template.report_template:
                 for res_id in template_res_ids:
+                    #Fix generate report for relevant partner only
+                    try:
+                        del ctx['active_ids']
```

Comment:
You should probably delete the key one level higher, before you go into the loop.